Add TraceMat method for finite field matrices

markuspf opened the pull request.

james-d-mitchell commented:
Can we have at least one test please @markuspf, for code coverage if nothing else?

james-d-mitchell commented:
What say you @wilfwilson ?

james-d-mitchell commented:
@markuspf can you also please rebase onto stable-3.0? Thanks!

wilfwilson commented:
I second your comments James, it would indeed be nice to have at least one test, and moreover thanks for doing this Markus.

markuspf commented:
Not entirely sure why github got its knickers in a twist over the second commit there?

markuspf commented:
I rebased on `stable-3.0` and added a test. I briefly thought about putting the `TraceMat` method in for semiring matrices in general, but then saw that that was more work than I thought...

wilfwilson commented:
Assuming that James can fix #508, then yes ;-)

Just want to reiterate that we really appreciate your help Markus. We try to be good `gap-system` citizens but you definitely make that easier.

james-d-mitchell commented:
Of course, I *can* fix it, but will I???

wilfwilson commented:
We can always just remove the ideals functionality from the Semigroups package ;-)

markuspf commented:
For reference: https://github.com/gap-system/gap/pull/2784
